db/inlineskiplist.h

```diff
@@ -188,6 +213,19 @@ class InlineSkipList {
   // level in [0..max_height_-1], if prev is non-null.
   Node* FindLessThan(const char* key, Node** prev = nullptr) const;
 
+  // Return the latest node with a key < key on stop_level. Start searching
+  // from top on the level below start_level.
+  // Fills prev[level] with pointer to previous node at "level" for every
+  // level in [stop_level..start_level-1], if prev is non-null.
+  Node* FindLessThan(const char* key, Node** prev, Node* top, int start_level,
+                     int stop_level) const;
+
+  // Fill hint->prev with pointer to previous node on levels in
+  // [stop_level, start_level]. Using previous value of hint->prev to help
+  // speed-up the search.
+  void FindWithHint(const char* key, InsertHint* hint, int start_level,
+                    int stop_level) const;
```

Comment:
It is still not clear to me what this function does after reading the comment. We can discuss offline.

Also maybe rename it to something like AdjustHintInPrev().